### PR TITLE
fix: support image cache on VFAT USB stick

### LINKED ISSUE #9936: detect FAT filesystem/volume labels

### PR DESCRIPTION
**Pull request by smira**
Scenario: copy contents of the ISO to the USB VFAT stick.

Make sure VFAT filesystem has a label `TALOS_*`.

Fixes #9936

### CODE REVIEW

**Comment by smira** [comment]
I got this fully tested e2e with the following flow:

* mount an ISO
* copy everything from an ISO to a "USB stick" FAT-formatted, with volume label `TALOS_*`
* boot Talos with USB stick mounted - image cache is correctly detected, and copied to the disk

File: internal/app/machined/pkg/controllers/block/internal/volumes/format.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ func Format(ctx context.Context, logger *zap.Logger, volumeContext ManagerContex
 		devPath = volumeContext.Status.Location
 	}
 
-	dev, err := blockdev.NewFromPath(devPath, blockdev.OpenForWrite())
+	dev, err := blockdev.NewFromPath(devPath)
```
Comment (smira) [inline internal/app/machined/pkg/controllers/block/internal/volumes/format.go:33]: dropped write bit here, as it's not needed, and breaks when the media is read-only (we're not writing to it anyways in this function)

File: internal/app/machined/pkg/controllers/cri/image_cache_config.go [hunk 2]
```diff
@@ -198,19 +199,26 @@ func (ctrl *ImageCacheConfigController) Run(ctx context.Context, r controller.Ru
 func (ctrl *ImageCacheConfigController) createVolumeConfigISO(ctx context.Context, r controller.ReaderWriter) error {
 	builder := cel.NewBuilder(celenv.VolumeLocator())
 
-	// volume.name == "iso9660" && volume.label.startsWith("TALOS_")
+	// volume.name in ["iso9660", "vfat"] && volume.label.startsWith("TALOS_")
```
Comment (smira) [inline internal/app/machined/pkg/controllers/cri/image_cache_config.go:202]: allow VFAT here to handle USB sticks

File: pkg/provision/providers/qemu/launch.go [hunk 3]
```diff
@@ -448,7 +448,7 @@ func launchVM(config *LaunchConfig) error {
 			)
 		case config.USBPath != "":
 			args = append(args,
-				"-drive", fmt.Sprintf("if=none,id=stick,format=raw,file=%s", config.USBPath),
+				"-drive", fmt.Sprintf("if=none,id=stick,format=raw,read-only=on,file=%s", config.USBPath),
```
Comment (smira) [inline pkg/provision/providers/qemu/launch.go:451]: make QEMU USB stick emulation read-only, so that we can mount same file to multiple VMs

**Review by frezbo — APPROVED** [review]

**Comment by rothgar** [comment]
Will there be another beta cut before the final release so I can test this or should I build from main?

**Comment by smira** [comment]
> Will there be another beta cut before the final release so I can test this or should I build from main?

There will be no other beta, but we have it fully tested. There's another bug fix in #9954

**Pull request merged**